File: bun-elysia/src/routes/hello.ts
import { Elysia, t } from "elysia";

export const helloRoutes = new Elysia({ prefix: "/api" })
  .get("/hello", () => ({ message: "Hello from Elysia!" }))
  .post(
    "/hello",
    ({ body }) => ({ message: `Hello, ${body.name}!` }),
    {
      body: t.Object({ name: t.String() }),
    }
  );

File: fastapi/app/routers/hello.py
from fastapi import APIRouter
from app.models.hello import HelloRequest, HelloResponse

router = APIRouter(prefix="/hello", tags=["hello"])

@router.get("/", response_model=HelloResponse)
async def get_hello():
    return HelloResponse(message="Hello from FastAPI!")

@router.post("/", response_model=HelloResponse)
async def post_hello(body: HelloRequest):
    return HelloResponse(message=f"Hello, {body.name}!")
